src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataControl.xaml.cs

```diff
@@ -2,76 +2,67 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.ComponentModel;
-using System.IO;
-using System.Linq;
-using System.Threading;
 using System.Windows;
 using System.Windows.Controls;
 using NuGet.PackageManagement.UI.ViewModels;
-using NuGet.Packaging;
 
 namespace NuGet.PackageManagement.UI
 {
 
     /// <summary>
     /// Interaction logic for PackageMetadataControl.xaml
     /// </summary>
-    public partial class PackageMetadataControl : UserControl
+    public partial class PackageMetadataControl : UserControl, IDisposable
     {
         public ReadMePreviewViewModel ReadMePreviewViewModel { get; set; }
 
+        private bool _initialVisibilitySet = false;
```

Comment:
I don't quite understand what this initialVisibilitySet is for. Could you leave a comment explaining its purpose? It looks like it's used to determine when to select the Readme tab. A comment would help explain what behavior this enforces.

Reply:
Hmm, before adding a comment, what I'm trying to do is make sure the right tab is selected when the control is first rendered. If there is a readme present then I was the ReadMe tab to be selected first. Otherwise I want to have the details tab selected. The problem I have is that on initial load of the component the DetailsModel doesn't have the package path set and so the ReadMe tab is disabled. Then it gets populated and the TabReadMe_IsVisibleChanged is fired. SO the first time it fires I want to set the selected tab correctly and I use this flag to determine if it's the first time that event is fired. 

If you have any suggestions let me know bc I don't love this solution. Otherwise I can add a comment explaining

Reply:
Is there an event when the package path gets set on the DetailsModel? Whenever the package path is populated, the UI could respond by setting the is visible on the Readme tab.